kraken/litmus/common_litmus.py

```diff
@@ -0,0 +1,92 @@
+
+    # wait until pods are running
+
+    ready_status = get_pod_ready_status()
+    logging.info("first ready status for pod" + str(ready_status))
+    while ready_status != "Running":
+        time.sleep(10)
+        logging.info("ready status for pod" + str(ready_status))
+        ready_status = get_pod_ready_status()
+        if "Error" in ready_status:
+            logging.error("Litmus pod had error creating")
+            sys.exit(1)
+
+
+# Check status of experiment
+def check_experiment(engine_name, experiment_name, namespace):
+    chaos_engine = runcommand.invoke("kubectl describe chaosengines/%s -n %s | grep "
+                                     "Status" % (engine_name, namespace))
+    engine_status = chaos_engine.split(": ")[-1].strip()
+    max_tries = 2
+    engine_counter = 0
+    while engine_status != "Running" and engine_status != "Completed":
+        time.sleep(10)
+        logging.info("Sleeping waiting for engine to start running")
+        chaos_engine = runcommand.invoke("kubectl describe chaosengines/%s -n %s | grep "
+                                         "Status" % (engine_name, namespace))
+        engine_status = chaos_engine.split(": ")[-1].strip()
+        if engine_counter >= max_tries:
+            logging.error("Chaos engine took longer than 5 minutes to be running or complete")
+            return False
+        engine_counter += 1
+        # need to see if error in run
+        if "not found" in engine_status:
+            return False
+
+    if not chaos_engine:
+        return False
+    chaos_result = runcommand.invoke("kubectl describe chaosresult %s"
+                                     "-%s -n %s | grep Verdict" %
+                                     (engine_name, experiment_name, namespace))
+    result_counter = 0
+    status = chaos_result.split(": ")[1].strip()
+    while status == "Awaited":
+        logging.info("Waiting for chaos result to finish waiting, sleeping 10 seconds")
+        time.sleep(10)
+        chaos_result = runcommand.invoke("kubectl describe chaosresult %s"
+                                         "-%s -n %s | grep Verdict" %
+                                         (engine_name, experiment_name, namespace))
+        status = chaos_result.split(": ")[1].strip()
+        if result_counter >= max_tries:
+            logging.error("Chaos results took longer than 5 minutes to get a final result")
+            return False
+        result_counter += 1
+        if "not found" in status:
+            return False
+
+    if status == "Pass":
+        return True
+    else:
+        return False
+
+
+# Delete all chaos engines in a given namespace
+def delete_experiments(namespace):
+    runcommand.invoke("kubectl delete chaosengine --all -n " + str(namespace))
+    runcommand.invoke("kubectl delete chaosexperiment --all -n " + str(namespace))
+    runcommand.invoke("kubectl delete chaosresult --all -n " + str(namespace))
```

Comment:
I think you forgot to cleanup service accounts here.

Reply:
The service accounts should be cleaned up here:  https://github.com/openshift-scale/kraken/blob/4b83f67379b058446b8fce7f5040ed61cbd43476/run_kraken.py#L285

Is that correct?